docs/1-tendermint-overview.md

```diff
@@ -0,0 +1,120 @@
+
+Tendermint is notable for its simplicity and performance and is currently one of the most mature and widely used implementations of BFT consensus.
+
+### Properties 
+
+The Tendermint consensus algorithm guarantees the following properties, with respect to a consensus instance:
+
+- **Agreement**: No two correct processes decide on different values
+
+- **Validity**: A value decided is a valid value, with respect to the specifications of validity of a message
+
+- **Termination**: All correct processes eventually decide
+
+- **Integrity**: No correct process decides more than once in a single height 
+
+In order to ensure all these properties, the Tendermint consensus algorithm assumes a correct majority of processes such that less than one third of processes are faulty (*f*):
+ 
+    n > 3f
+ 
+When a process *p* is faulty, *p* can behave in an arbitrary way and we have no assumptions or guarantees regarding its actions.
+ 
+In order to simplify the whole discussion, we assume the following from now on:
+ 
+    n = 3f + 1
+
+### Overview
+
+The consensus algorithm proceeds in rounds: in each round there is a validator that proposes a value (proposer) and all the validators then vote during the round on whether to accept the proposed value or move on to the next round (proposers are chosen according to their voting power).
+
+During a round, processes can exchange different types of messages:
+ 
+- **PROPOSAL message**: sent by the proposer of the round to propose a value in a new round
+
+- **PREVOTE message**: sent by validators to vote for a value during the first voting step
+
+- **PRECOMMIT message**: sent by validators to vote for a value during the second voting step
+ 
+We can summarize the execution in a round in **two voting steps**: PREVOTE and PRECOMMIT steps. A vote can be for a particular value or for *nil* (null value).
+
+A correct process decides on a value *v* in a round *r* upon receiving a proposal and *2f + 1* quorum of PRECOMMIT messages for *v* in a round *r*. 
+A correct process sends a PRECOMMIT message for a value *v* in a round *r* upon receiving a proposal and *2f + 1* quorum of PREVOTE messages for *v* in a round *r*.
+
+Validators wait some time before sending a PREVOTE for *nil* if they do not receive a valid proposal after a certain time (*proposalTimeout*) and they send a PRECOMMIT message for *nil* if they do not receive *2f + 1* matching PREVOTE messages for a value after a certain time (*prevoteTimeout*).
+If processes do not receive *2f + 1* matching PRECOMMIT messages for a value after a certain time (*precommitTimeout*), they move to the next round.
+
+After a decision has been made, processes continue to agree on other values on another consensus instance (*height*) and they repeat the process described above in order to agree on different transactions.
+
+### Rules
```

Comment:
Maybe "Correct behaviour"?

Reply:
Yes, "Correct behavior" would be good as well. I named this sub-section "Rules" because I wanted to give a simple overview of the constraints used to guarantee the 4 properties. But I agree that correct behavior gives a better idea